examples/nidcpower/measure-record.py

```diff
@@ -224,9 +201,13 @@ def _on_close(event):
         error_message = (
             "The operation is not implemented or is not supported/enabled in this service"
         )
+    elif rpc_error.code() == grpc.StatusCode.UNKNOWN:
+        # The error details is a JSON string containing both the status code and the description.
```

Comment:
Is that guaranteed? Does gRPC itself ever return `StatusCode.UNKNOWN` with a non-JSON message?

Reply:
> My earlier searching said the server never returned it, but apparently it can, at least in theory:
> [Server side application throws an exception (or does something other than returning a Status code to terminate an RPC)](https://grpc.github.io/grpc/core/md_doc_statuscodes.html)

Yeah, it's definitely possible for the client to return `StatusCode.UNKNOWN` when the server returns an unexpected response:
- https://github.com/grpc/grpc/blob/master/src/core/lib/surface/call.cc#L983
- https://github.com/grpc/grpc/blob/master/src/core/lib/transport/status_conversion.cc#L67

Even if `ni_grpc_device_server.exe` never hits any of these cases, a more complicated setup like an HTTP load balancer or proxy server could conceivably hit one of this cases.

Compared to the code you removed and the rest of the error handling code, adding a try/except doesn't seem that bad to me.

```
    elif rpc_error.code() == grpc.StatusCode.UNKNOWN:
        try:
            error_details = json.loads(error_message);
            error_message = f"Failed with status {error_details['code']}\n{error_details['message']}"
        except JSONDecodeError:
            pass
```

> I don't see any other good options. 

Custom trailing metadata or "richer error model" (`google.rpc.Status`) would not have this ambiguity. :)